lightning/src/blinded_path/payment.rs

```diff
@@ -0,0 +1,357 @@
+//! Data structures and methods for constructing [`BlindedPath`]s to send a payment over.
+//!
+//! [`BlindedPath`]: crate::blinded_path::BlindedPath
+
+use bitcoin::secp256k1::{self, PublicKey, Secp256k1, SecretKey};
+use crate::blinded_path::BlindedHop;
+use crate::blinded_path::utils;
+use crate::io;
+use crate::ln::PaymentSecret;
+use crate::ln::features::BlindedHopFeatures;
+use crate::ln::msgs::DecodeError;
+use crate::offers::invoice::BlindedPayInfo;
+use crate::prelude::*;
+use crate::util::ser::{Readable, Writeable, Writer};
+
+/// Data to construct a [`BlindedHop`] for sending a payment over.
+///
+/// [`BlindedHop`]: crate::blinded_path::BlindedHop
+pub enum BlindedPaymentTlvs {
+	/// This blinded payment data is to be forwarded.
+	Forward {
+		/// The short channel id this payment is being forwarded over.
+		short_channel_id: u64,
+		/// Payment parameters for relaying over this channel.
+		payment_relay: PaymentRelay,
+		/// Payment constraints when relaying over this channel.
+		payment_constraints: PaymentConstraints,
+		/// Supported and required features when relaying a payment onion containing this object's
+		/// corresponding [`BlindedHop`].
+		///
+		/// [`BlindedHop`]: crate::blinded_path::BlindedHop
+		features: BlindedHopFeatures,
+	},
+	/// This blinded payment data is to be received.
+	Receive {
+		/// Used to authenticate the sender of a payment to the receiver and tie MPP HTLCs together.
+		payment_secret: PaymentSecret,
+		/// Constraints for the receiver of this payment.
+		payment_constraints: PaymentConstraints,
```

Comment:
Discussed offline, gonna keep this field for the sake of the `htlc_minimum_msat` field. Updated #1970 because we should fuzz that field in the future for anti-probing reasons.